src/transformState.js

```diff
@@ -1,11 +1,30 @@
-'use strict';
+"use strict";
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case "addProperties":
+        Object.assign(state, extraData);
+        break;
+
+      case "removeProperties":
+        for (const key of keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case "clear":
```

Comment:
However, there is a small issue with the "clear" case. When you want to clear all properties from the state object, using a for...in loop may not work as expected. Instead, you should use Object.keys to get an array of keys and then delete each key.